generator/codegen/common.go

```diff
@@ -169,3 +170,11 @@ func AddValuesToParams(d *Data, getValues func(*openapi3.T) (interface{}, error)
 	d.Params = params
 	return nil
 }
+
+func Map[T, V any](input []T, mapfn func(T) V) []V {
```

Comment:
I think this is overkill, you are using it in one place so it seems a bit premature to be creating a generic function for mapping.

Also input can be `nil` which means `len(input)` would be 0 (special case of nil slices) which means you would allocate an empty slice if you passed in nil which I wouldn't expect in a map function.

Also you didn;t check if the function is nil which would panic.

But generally I think this is a bit premature unless we were doing a lot of mappings